core/src/test/java/org/infinispan/conflict/impl/ConflictManagerTest.java

```diff
@@ -277,21 +266,30 @@ private void dropClusteredGetCommands() {
       IntStream.range(0, numMembersInCluster).forEach(i -> wrapInboundInvocationHandler(getCache(i), DropClusteredGetCommandHandler::new));
    }
 
-   private void delayStateTransferCompletion() {
-      IntStream.range(0, numMembersInCluster).forEach(i -> wrapInboundInvocationHandler(getCache(i), DelayStateRequestCommandHandler::new));
+   private void delayStateTransferCompletion(CountDownLatch latch) {
+      IntStream.range(0, numMembersInCluster).forEach(i -> wrapInboundInvocationHandler(getCache(i), delegate -> new DelayStateResponseCommandHandler(latch, delegate)));
    }
 
-   private class DelayStateRequestCommandHandler implements PerCacheInboundInvocationHandler {
+   private class DelayStateResponseCommandHandler implements PerCacheInboundInvocationHandler {
+      final CountDownLatch latch;
       final PerCacheInboundInvocationHandler delegate;
 
-      DelayStateRequestCommandHandler(PerCacheInboundInvocationHandler delegate) {
+      DelayStateResponseCommandHandler(CountDownLatch latch, PerCacheInboundInvocationHandler delegate) {
+         this.latch = latch;
          this.delegate = delegate;
       }
 
       @Override
       public void handle(CacheRpcCommand command, Reply reply, DeliverOrder order) {
          if (command instanceof StateResponseCommand) {
-            TestingUtil.sleepThread(STATE_TRANSFER_DELAY);
+            StateResponseCommand stc = (StateResponseCommand) command;
+            boolean isLastChunk = stc.getStateChunks().stream().anyMatch(StateChunk::isLastChunk);
+            if (isLastChunk) {
+               try {
+                  latch.await();
```

Comment:
I fear this will hang during teardown if the test fails and doesn't call `latch.countDown()`. I have some thoughts about making it easier to close latches in tests, but in the meantime I suggest adding a timeout here.